### PR TITLE
Added blue-green deployment scripts

### PR DESCRIPTION
**Pull request by mauriciofreshw**
This is a extra workflow attached to the tag trigger that will clone a current environment ( enhanced-travel-screening-dev-2) and deploy to it.

### CODE REVIEW

**Comment by fwkendall** [comment]
This looks great :) I don't think it's ready to merge in yet though:

1. The yaml looks totally specific to dev, is it possible to get things keying off the current environment?
2. It looks like the GH Action being used is not in this repo still, right?

**Comment by mauriciofreshw** [comment]
> This looks great :) I don't think it's ready to merge in yet though:
> 
> 1. The yaml looks totally specific to dev, is it possible to get things keying off the current environment?
> 2. It looks like the GH Action being used is not in this repo still, right?

1. It is. Case we need to use this on prod we just need to clone the current prod environment, add the security group and then point the DEPLOY_ENV accordingly. I didn't set to prod yet because I would like to test this on dev first.

2. You mean the **actions-aws-eb@v1.0.0**? Not yet, but we should if everything goes well (Does this requires high-level permissions from the bcgov acc?).

**Comment by fwkendall** [comment]
> 1. It is. Case we need to use this on prod - and will - we just need to clone the current prod environment, add the security group and then point the DEPLOY_ENV accordingly. I didn't set to prod yet because I would like to test this on dev first.
Gotcha. I noticed that lines 1, 6, and 12 in the yaml all reference dev (although two of those are descriptors). So maybe we just need to update those then?

> 2. You mean the **actions-aws-eb@v1.0.0**? Not yet, but we should if everything goes well (Does this requires high-level permissions from the bcgov acc?).
Yep that's the one I mean. :) I don't think it requires anything other than access to our own repo (as far as I'm aware). So the process we're going to be following here is deploy+test in dev, then once we validate there make the changes necessary and redeploy?

**Comment by mauriciofreshw** [comment]
> Gotcha. I noticed that lines 1, 6, and 12 in the yaml all reference dev (although two of those are descriptors). So maybe we just need to update those then?
> 
> Yep that's the one I mean. :) I don't think it requires anything other than access to our own repo (as far as I'm aware). So the process we're going to be following here is deploy+test in dev, then once we validate there make the changes necessary and redeploy?

Once we validate then we change those first lines to a push trigger like this, assuming we want to deploy on merging to master:

`on:
  push:
    branches: [ master ]`

and also change all necessary names/variables to prod.

File: .github/workflows/build-blue-green.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        run: echo "##[set-output name=env;]$(make gh-get-current-aws-env)"
+        id: current_env_name
+
+      - name: Extract current date
+        shell: bash
+        run: echo "##[set-output name=date;]$(date '+%Y%m%d%H%M')"
+        id: extract_date
+
+      - name: Clone EB enviroment
+        uses: mauriciofreshw/actions-aws-eb@v1.0.0
```
Comment (ashtonmeuser) [inline .github/workflows/build-blue-green.yml:61]: We should be able to replace this with a standard GH Action container. See [this](https://github.com/actions/setup-python) for running Python. Then all we would need is to `pip install awsebcli`.

Comment (mauriciofreshw) [reply .github/workflows/build-blue-green.yml]: True. I changed the script and tested on https://github.com/mauriciofreshw/eb-blue-green-test/actions/runs/125227867/workflow

thanks!

**Comment by fwkendall** [comment]
This is looking good to go for dev, I think the only thing remaining is to port over the image promotion from https://github.com/bcgov/Foreign-Farm-Workers-Permitting-System/pull/43/

**Pull request merged**